test/tests/resolver_tests.py

```diff
@@ -466,3 +471,64 @@ def test_env_order(self):
                              "Got the following instead: \n{}"
                              .format(''.join(["{}: {}\n".format(*v) for v in
                                               exports])))
+
+    def test_version_compatibility(self):
+        """Make sure version compatibility checks are working and populate the
+        results.json file correctly."""
+
+        pav_version = PavVars.version(self)
+
+        arg_parser = arguments.get_parser()
+        args = arg_parser.parse_args([
+            'run',
+            '-w',
+            '5',
+            'version_compatible'
+        ])
+
+        expected_results = {
+            'version_compatible.one': {
+                'test_version': '1.2.3',
+                'pav_version': pav_version,
+                'compatible_pav_versions': 'none'
+            },
+            'version_compatible.two': {
+                'test_version': 'beta',
+                'pav_version': pav_version,
+                'compatible_pav_versions': '1.2.3-5.4.9'
+            },
+            'version_compatible.three': {
+                'test_version': '1.0',
+                'pav_version': pav_version,
+                'compatible_pav_versions': 'none'
+            }
+        }
+
+        # Ensures Version information gets populated correclty even with empty
+        # version section in test config
+        run_cmd = commands.get_command(args.command_name)
```

Comment:
There's are useful .silence() and .clear_output() methods on all commands that do this setup now.